src/auth/auth.service.ts

```diff
@@ -121,19 +123,33 @@ export class AuthService {
   async verifyAccountActivationToken(
     jwtToken: string,
   ): Promise<{ id: number }> {
+    const invalidTokenMessage =
+      'Invalid token. Please provide a valid token to activate account';
+
+    if (!isJwtTokenStructureValid(jwtToken)) {
+      throw new UnauthorizedException(invalidTokenMessage);
+    }
```

Comment:
Do we need this check? `jwtService.verifyAsync` should be enough as it also checks the token structure